### PR TITLE
Support default dtype in `L.ResNetLayers`

### PR DESCRIPTION
**Pull request by takagi**
This PR is a part of #4582, makes `L.ResNetLayers` use the default dtype in its helper method.

### CODE REVIEW

File: tests/chainer_tests/links_tests/model_tests/test_vision.py [hunk 1]
```diff
@@ -12,14 +13,24 @@
...
+    'dtype': numpy.float16,
+    'n_layers': 50,
+    'downsample_fb': False,
+}])
 @unittest.skipUnless(resnet.available, 'Pillow is required')
 @attr.slow
 class TestResNetLayers(unittest.TestCase):
 
     def setUp(self):
+        self._old_dtype = None
```
Comment (okuta) [inline tests/chainer_tests/links_tests/model_tests/test_vision.py]: Maybe you can use `getattr`.
```self._old_dtype = getattr(config._local, 'dtype', None)```

**Comment by okuta** [comment]
jenkins, test this please.

File: tests/chainer_tests/links_tests/model_tests/test_vision.py (hunk 1 above)
Comment (takagi) [reply tests/chainer_tests/links_tests/model_tests/test_vision.py]: Fixed!

File: tests/chainer_tests/links_tests/model_tests/test_vision.py [hunk 2]
```diff
@@ -12,14 +13,22 @@
...
+    'dtype': numpy.float16,
+    'n_layers': 50,
+    'downsample_fb': False,
+}])
 @unittest.skipUnless(resnet.available, 'Pillow is required')
 @attr.slow
 class TestResNetLayers(unittest.TestCase):
 
     def setUp(self):
+        config = chainer.config
```
Comment (beam2d) [inline tests/chainer_tests/links_tests/model_tests/test_vision.py]: See this comment: https://github.com/chainer/chainer/pull/5114#discussion_r203917883

Comment (takagi) [reply tests/chainer_tests/links_tests/model_tests/test_vision.py]: Fixed!

**Comment by takagi** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**